Cubical/Foundations/Isomorphism.agda

```diff
@@ -0,0 +1,94 @@
+{-
+
+Theory about isomorphisms
+
+- Definitions of [section] and [retract]
+- Definition of isomorphisms ([Iso])
+- Any isomorphism is an equivalence ([isoToEquiv])
+
+-}
+{-# OPTIONS --cubical --safe #-}
+module Cubical.Foundations.Isomorphism where
+
+open import Agda.Primitive
+
+open import Cubical.Core.Everything
+open import Cubical.Foundations.HLevels
+
+-- Section and retract
+module _ {ℓ ℓ'} {A : Set ℓ} {B : Set ℓ'} where
+  section : (f : A → B) → (g : B → A) → Set ℓ'
+  section f g = ∀ b → f (g b) ≡ b
+
+  -- NB: `g` is the retraction!
+  retract : (f : A → B) → (g : B → A) → Set ℓ
+  retract f g = ∀ a → g (f a) ≡ a
+
+
+record isoStruct {ℓ ℓ'} {A : Set ℓ} {B : Set ℓ'} (f : A → B) : Set (ℓ ⊔ ℓ') where
+  field
+    inverse : B → A
```

Comment:
Why not call this simply `inv`? Then it would match the other projections better

Reply:
Ok.

Reply:
You say we call the maps "f" and "inv"?

Reply:
What about `fun` and `inv`? (that makes it all a bit funnier :-) )

Reply:
Not sure about a good name for the map. Maybe "map", since it is the map?

Reply:
Ok;) Let's go about it the funny way.

Reply:
Agda doesn't know how to parse with the _,_,_,_. I switch to iso.